src/Microsoft.Health.Fhir.Core/Microsoft.Health.Fhir.Core.csproj

```diff
@@ -44,6 +44,7 @@
   </ItemGroup>
 
   <ItemGroup>
+    <ProjectReference Include="..\Microsoft.Health.CosmosDb\Microsoft.Health.CosmosDb.csproj" />
```

Comment:
Core shouldn't reference CosmosDb

Reply:
Yes.  This was something I did try, but I expected this change to get reverted when I reverted to  Commit #3 to put the exception back in it's original location.  I'm not sure why it didn't.  Investigating...

Reply:
I've solved this by moving KnownCosmosDbCmkSubStatusValue into Core\Exceptions, so it lives alongside the CustomerManagedKeyException that leverages it.  The downside here is that we now have Cosmos DB related "info" in Core.  But, I couldn't find a better location that avoided the circular reference.  I'm open to suggestions.

Reply:
Based on an offline suggestion from @namadabu , I've moved KnownCosmosDbCmkSubStatusValue and the calculation of the BYOK error message into Microsoft.Health.Fhir.CosmosDb.  This removes any Cosmos-related knowledge from Core